### ISSUE TITLE
haddock run as part of new-build ignores internal libraries

### ISSUE THREAD

**Issue by Mikolaj** [haskell/cabal#5949]
When I run 'cabal new-build', haddock ignores internal libraries. E.g., in this project

https://github.com/LambdaHack/LambdaHack/blob/ed0328a231a61adfe551de78c4814a9565a7681a/LambdaHack.cabal#L164

OTOH, 'cabal new-haddock' processes all the internal libraries.

BTW, the internal libraries are very useful to me; in particular, they lower compilation time considerably (.hi files don't hog gigabytes of memory).

**Comment by ezyang** [comment]
Yes, I think this is a case where someone just needs to look into what actually needs to happen to implement it. When we initially added internal libraries we didn't think to carefully about the interaction with Haddock.

**Comment by Mikolaj** [comment]
Just to confirm the problem is still there with cabal from master branch. However, much more pressing are haddock-only (probably) bugs with internal libraries, see https://github.com/haskell/haddock/issues/1363

**Comment by Mikolaj** [comment]
Progress on the haddock front: https://github.com/haskell/haddock/pull/1277 https://github.com/haskell/haddock/pull/1379

What is still missing in cabal? Are some manual workarounds now possible using the new functionality in haddock?
